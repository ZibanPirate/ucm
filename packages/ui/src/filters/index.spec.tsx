import { fireEvent, render, screen } from "@testing-library/react";

import { Filter, Filters } from ".";

describe(`Testing component '${Filters.name}' :`, () => {
  it(`should render with options of all types, then change their values`, () => {
    const filters: Filter[] = [
      {
        label: "Make",
        name: "make",
        type: "options",
        options: [
          { name: "BMW", checked: true },
          { name: "Porsche", checked: false },
        ],
      },
      {
        label: "Price",
        name: "price",
        type: "range",
        options: {
          min: 20000,
          max: 100000,
        },
      },
    ];

    const { container } = render(<Filters filters={filters} />);

    const bmwOption = screen.getByText("BMW");
    fireEvent.click(bmwOption);
    const priceRangeMin = screen.getByDisplayValue("20000");
    fireEvent.change(priceRangeMin, { target: { value: 20001 } });
    const priceRangeMax = screen.getByDisplayValue("100000");
    fireEvent.change(priceRangeMax, { target: { value: 100001 } });

    expect(container).toMatchSnapshot();
  });
});
