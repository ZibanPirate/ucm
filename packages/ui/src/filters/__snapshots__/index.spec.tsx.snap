// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'Filters' : should render with options of all types, then change their values 1`] = `
<div>
  <div
    style="padding: 0px 1rem 1rem;"
  >
    <div
      style="padding-top: 1rem;"
    >
      <div
        style="text-align: center;"
      >
        <span
          style="font-size: large; margin: 0px 0px .5rem; display: flex;"
        >
          Make
        </span>
      </div>
      <div>
        <div>
          <input
            checked=""
            id="BMW"
            name="BMW"
            type="checkbox"
          />
          <label
            for="BMW"
          >
            BMW
          </label>
        </div>
        <div>
          <input
            id="Porsche"
            name="Porsche"
            type="checkbox"
          />
          <label
            for="Porsche"
          >
            Porsche
          </label>
        </div>
      </div>
    </div>
    <div
      style="padding-top: 1rem;"
    >
      <div
        style="text-align: center;"
      >
        <span
          style="font-size: large; margin: 0px 0px .5rem; display: flex;"
        >
          Price
        </span>
      </div>
      <div>
        <div>
          <input
            id="Min"
            name="Min"
            type="number"
            value="20000"
          />
          <label
            for="Min"
          >
            Min
          </label>
        </div>
        <div>
          <input
            id="Max"
            name="Max"
            type="number"
            value="100000"
          />
          <label
            for="Max"
          >
            Max
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
